clients/src/main/java/org/apache/kafka/common/errors/PendingOffsetException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.errors;
+
+/**
+ * Exception thrown when there are pending offsets for the requested topic partitions.
+ */
+public class PendingOffsetException extends RetriableException {
```

Comment:
Perhaps `PendingOffsetCommitException` (and similarly `Errors.PENDING_OFFSET_COMMIT`)?

Reply:
I think PendingOffset is okay, also there is a `PendingCommittedOffset` struct in the consumer coordinator, which is somehow a little bit overlapping with the suggested term.

Reply:
The problem is it doesn't suggest the context of the consumer's offset commits. The name "pending offset" could apply to appends as well. If the concern is overlap with `PendingCommittedOffset`, we could rename that class to `InFlightOffsetCommit` which is more suggestive anyway.

Reply:
An alternative to consider is `UNSTABLE_OFFSET_COMMIT`. This would make it more symmetric with the `requireStable` flag.